slack/handler.go

```diff
@@ -83,25 +79,83 @@ func (r *rtmBot) handleEvent(msg slack.RTMEvent) error {
 		return errors.New("Invalid auth received")
 	case *slack.HelloEvent:
 		r.logger.Println("Received hello, sending greetings !")
-		r.rtm.SendMessage(r.rtm.NewOutgoingMessage("Hello, I'm musicof, let's play !", r.channel.ID))
+		r.rtm.PostMessage(r.channel.ID, "Hello, I'm musicof, let's play !", slack.PostMessageParameters{})
 	case *slack.ConnectedEvent:
 		r.logger.Println("Connected !")
 	case *slack.MessageEvent:
 		if err := r.handleMessage(ev); err != nil {
+			r.logger.Println("Failed to handle message, reason :", err)
 			return err
 		}
 	}
 
 	return nil
-
 }
 
-func (r *rtmBot) handleHalt() {
-	r.rtm.SendMessage(r.rtm.NewOutgoingMessage("See you later !", r.channel.ID))
+func (r *rtmBot) handleHalt() error {
+	r.rtm.PostMessage(r.channel.ID, "See you later !", slack.PostMessageParameters{})
+	return r.rtm.Disconnect()
 }
 
 func (r *rtmBot) handleMessage(ev *slack.MessageEvent) error {
-	// TODO write something interesting
+	if ev.Channel != r.channel.ID {
+		return nil
+	}
 
-	return nil
+	if !strings.Contains(ev.Text, r.rtm.GetInfo().User.ID) {
+		return nil
+	}
+
+	if !strings.Contains(ev.Text, "nominate") {
+		return nil
+	}
+
+	return r.handleNominate(ev.User)
+}
+
+func (r *rtmBot) handleNominate(callerID string) error {
+	botID := r.rtm.GetInfo().User.ID
+	unwantedIDs := []string{botID, callerID}
+
+	usersInConversationParameters := &slack.GetUsersInConversationParameters{ChannelID: r.channel.ID}
+	userIDs, _, err := r.rtm.GetUsersInConversation(usersInConversationParameters)
+
+	if err != nil {
+		return err
+	}
+
+	userIDs = remove(userIDs, unwantedIDs)
+
+	userID := userIDs[rand.Intn(len(userIDs))]
+
+	user, err := r.rtm.GetUserInfo(userID)
+
+	if err != nil {
+		return err
+	}
+
+	_, _, err = r.rtm.PostMessage(r.channel.ID, "@"+user.Name, slack.PostMessageParameters{LinkNames: 1})
+
+	return err
+}
+
+func remove(s []string, r []string) []string {
+	for i := 0; i < len(s); i++ {
+		if find(s[i], r) {
+			s = append(s[:i], s[i+1:]...)
```

Comment:
What happens when i == len(s-1) ?

Reply:
I would prefer to define a new "result slice", and use it as an accumaltor.

Actually I'm not sure to understand your code right now...

For instance 

```
func filter(in []string, excludedValues ...string) []string {
    res := make([]string, len(in))
    for _, item := range in {
        if find(item, excludedValues) {
            continue
        }

        res = append(res, value)
    }
}
```

Reply:
We also need to write some tests about those `methods` find and `filter`